mod/reverse.go

```diff
@@ -67,11 +66,23 @@ func (r *Reverser) Write(raw map[string]interface{}) error {
 		}
 
 		createdFile := strKey + ext
-
-		err := r.LuaWriter.EncodeToFile(strVal, createdFile)
-		if err != nil {
-			return fmt.Errorf("lua.EncodeToFile(<value>, %s) : %v", createdFile, err)
+		
+		var jsonInterface map[string]interface{}
+		err := json.Unmarshal([]byte(strVal), &jsonInterface)
+		if err == nil {
+			// if it's JSON, use the JSONWriter
+			err := r.ObjWriter.WriteJSON(jsonInterface, createdFile)
```

Comment:
This does more than format. You're writing a new file? How often is the script state illegible as a string?

Reply:
Script state as string is pretty much always illegible and it also sucks for source control since it's a single line. See here for a recent example: https://github.com/argonui/SCED/pull/786/files#diff-34b51a0c82f85d6d9927ddc988df3589947b8c241d04199af6a4dfbc3c3f6044